annotated.go

```diff
@@ -1300,7 +1317,11 @@ func (at *asAnnotation) results(ann *annotated) (
 
 		newOutResult := reflect.New(resType).Elem()
 		for i := 1; i < resType.NumField(); i++ {
```

Comment:
Is there any reason we would need to check `Value.CanConvert` first since technically `Value.Convert` can panic even if `Type.ConvertibleTo` returns true? I can't think of any reason why a func value wouldn't be convertible if its type is though.